semantic/utilities.go

```diff
@@ -0,0 +1,30 @@
+package semantic
+
+import (
+	"fmt"
+	"math/big"
+)
+
+// convertToBigInt attempts to convert the given str to a big.Int,
+// returning an error if the conversion fails.
+//
+// For convenience, it also returns a boolean indicating whether the given
+// string was a number or not, which can be useful when the string not being
+// a number is actually possible.
+func convertToBigInt(str string) (res *big.Int, err error, isNumber bool) {
```

Comment:
One more thing here - it looks like isNumber is always true if err is nil. I think it'd be cleaner to check for err != nil when we want to find out if this was a number so we don't need an extra return value.

Reply:
Yes that's what I explained [in this comment](https://github.com/google/osv-scalibr/pull/316#discussion_r1907644492), which [you said you could see the benefit in](https://github.com/google/osv-scalibr/pull/316#discussion_r1922697256).

Are you saying you've changed your mind, or misunderstood the original thread?

Reply:
Ah, I thought in the other case we'd construct some sort of special error ourselves if v is not a number.

I'd say we can just return an error and use that to check that the arg was a number. You don't need to do
```
vv, vIsNumber := convertToBigInt(vt.value)
if vIsNumber != nil {
	...
}
```
You can just call the error `err` and add a comment that the input wasn't a number if there's an error parsing it.